### PR TITLE
slim docker image

### PR DESCRIPTION
**Pull request by ocefpaf**
@lwasser this is a working in progress. I have a few ideas that can give you a more stable environment and lighter docker image.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
-FROM continuumio/anaconda3
+FROM continuumio/miniconda3:4.6.14
```
Comment (ocefpaf) [inline Dockerfile]: @lwasser the main difference is here. We are better off with a light miniconda rather than the full Anaconda distro.

File: Dockerfile [hunk 2]
```diff
@@ -16,20 +17,15 @@ COPY setup-r-envt.R setup-r-envt.R
...
-RUN conda config --add channels conda-forge
-
-RUN conda config --set channel_priority strict
-
 COPY environment.yml environment.yml
-
-RUN conda update -n base -c defaults conda
-
-RUN conda env update -n root -f environment.yml
+RUN conda config --add channels conda-forge \
```
Comment (ocefpaf) [inline Dockerfile]: We should try to put as many docker commands as we can into a single one to avoid bloating the docker image. But this change is not related to the problem we are trying to solve with the env.

File: Dockerfile [hunk 3]
```diff
@@ -16,20 +17,15 @@ COPY setup-r-envt.R setup-r-envt.R
...
+    && conda config --set channel_priority strict \
+    && conda update --all --yes \
+    && conda env update -n base -f environment.yml \
+    && conda clean --all --yes --force-pkgs-dirs
```
Comment (ocefpaf) [inline Dockerfile]: `--all --yes` is = `tipsy` and `--force-pkgs-dirs` get us some extra free space.

File: Dockerfile (hunk 2 above)
Comment (lwasser) [reply Dockerfile]: wow - i didn't know that streamlines the build!! thank you @ocefpaf

**Comment by ocefpaf** [comment]
I got a successful docker build with this locally. If you want to merge go ahead. I'll play with unpinned and using conda's `R` to avoid the long R build/install times there. (I'm not sure if we have all the r-packages but we can still do a `install.packages` or just add them to conda-forge.)

**Comment by lwasser** [comment]
hey @ocefpaf just a note that this container built for me perfectly and QUICKLY just now!! i'll merge this.

**Comment by ocefpaf** [comment]
> hey @ocefpaf just a note that this container built for me perfectly and QUICKLY just now!! i'll merge this.

I have a few meeting now but I'll send another PR later to play with the R install. Hopefully we can make that build quicker and the image smaller.

**Comment by lwasser** [comment]
ok awesome. @ocefpaf i'll merge this and we can play with r later. I think the one package we had a hard time with via conda for R was rsaga.
